linkerd/core/src/test/scala/io/buoyant/linkerd/RequestEvaluatorTest.scala

```diff
@@ -0,0 +1,106 @@
+package io.buoyant.linkerd
+
+import com.twitter.finagle._
+import com.twitter.finagle.buoyant.Dst
+import com.twitter.finagle.client.Transporter.EndpointAddr
+import com.twitter.finagle.context.Contexts
+import com.twitter.finagle.http.{MediaType, Request, Response}
+import com.twitter.finagle.naming.NameInterpreter
+import com.twitter.finagle.naming.buoyant.DstBindingFactory
+import com.twitter.io.Buf
+import com.twitter.util.{Activity, Future, Var}
+import io.buoyant.config.Parser
+import io.buoyant.namer.{DelegateTree, Delegator}
+import io.buoyant.router.RoutingFactory.BaseDtab
+import io.buoyant.router.context.{DstBoundCtx, DstPathCtx}
+import io.buoyant.test.FunSuite
+
+class EvaluatorNamer(delegation: DelegateTree[Name.Bound]) extends NameInterpreter with Delegator {
+  def delegate(
+    dtab: Dtab,
+    tree: NameTree[Name.Path]
+  ): Future[DelegateTree[Name.Bound]] = Future.value(delegation)
+
+  def dtab: Activity[Dtab] = Activity.pending
+
+  override def bind(
+    dtab: Dtab,
+    path: Path
+  ): Activity[NameTree[Name.Bound]] = Activity.pending
+}
+
+class RequestEvaluatorTest extends FunSuite {
+  val jsonMapper = Parser.jsonObjectMapper(Nil)
+
+  val successMessage = "request succeeded"
+  val testService = Service.mk[Request, Response] { req =>
+    val rsp = Response()
+    rsp.content = Buf.Utf8(successMessage)
+    Future.value(rsp)
+  }
+
+  val testStack = DstPathCtx.Setter.module.toStack(
```

Comment:
I think this can be
`DstPathCtx.Setter.module +: RequestEvaluator.module +: Stack.Leaf(...)`

Reply:
Agreed this is much clearer.